### ISSUE TITLE
[research 1d]: can we decommission the CSW service?

### ISSUE THREAD

**Issue by adborden** [GSA/data.gov#2383]
## User Story

In order to reduce compliance overhead and operations and maintenance costs for data.gov and redirect resources on higher value services, the data.gov team wants to decommission the CSW service.

## Acceptance criteria
- [ ] We can concretely describe who is using the CSW service and how much
- [ ] We have identified whether there is at least one potential alternate or reduced O&M implementation
- [ ] There's an issue in the backlog for followup based on the findings

## Background
[CSW issues in GitHub](https://github.com/GSA/datagov-deploy/issues?q=is%3Aissue+is%3Aopen+label%3Acomponent%2Fcsw).

- We don't have great insight into the CSW service. We know there are issues with how it loads datasets from CKAN and don't have much confidence that the data in CSW is accurate or complete.
- CSW is not core to the data.gov mission.
- We don't know how to filter datasets in CSW to only geospatial datasets which would allow GeoPlatform to use it for harvesting. GeoPlatform has also indicated they are looking at harvesting from the CKAN API instead.

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))
[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
This would reduce the complexity of Data.gov by reducing number of hosts, reducing services, and reducing attack surface.

## Sketch

I don't think CSW endpoints are captured in any analytics. We can review the pycsw and access logs to see how frequently it is being used. CSW supports queries in POST requests, so it might be difficult to determine what queries are being used.

**Comment by adborden** [comment]
pycsw access logs are baked in with the ckan access logs. I pulled the last 30 days:
```
dsh -g pycsw-web-v1 sudo zgrep -h -i /csw /var/log/ckan/ckan.access.log.?.gz  > pycsw.log
dsh -g pycsw-web-v1 sudo zgrep -h -i /csw /var/log/ckan/ckan.access.log.1?.gz  >> pycsw.log
dsh -g pycsw-web-v1 sudo zgrep -h -i /csw /var/log/ckan/ckan.access.log.2?.gz  >> pycsw.log
dsh -g pycsw-web-v1 sudo zgrep -h -i /csw /var/log/ckan/ckan.access.log.30.gz  >> pycsw.log
```

I shove this into a rough [script](https://gist.github.com/adborden/dea78eb69e344e7a892ef927b5f4c5b3) to normalize the request parameters.

Over the course of 30 days: 37168 requests from 2021-01-02 06:41:37 to 2021-02-01 06:49:05


I removed or filtered out:
- anything from an internal IP (netscaler healtcheck)
- request=getcapabilities service=csw version=2.0.2 (our NR ping check)
- a handful of burp/fuzzing/pen testing

This leaves us with approximately 1200 requests over 30 days across 334 unique IP addresses.
https://docs.google.com/spreadsheets/d/1Y96tKFj00R5iEy1HGlSALrceU2-aWi6p3DDoW5EgzkI/edit#gid=1543708280
